src/Farmer/Types.fs

```diff
@@ -13,10 +13,10 @@ type ResourceName =
         | r -> r
     member this.Map mapper = match this with ResourceName r -> ResourceName (mapper r)
 
-/// An Azure ARM resource
-type IResource =
+/// An Azure ARM resource value, which can be serialized to JSON.
+type IArmResource =
     abstract member ResourceName : ResourceName
-    abstract member ToArmObject : unit -> obj
+    abstract member JsonValue : obj
```

Comment:
I am being pedantic here, so really I am fine with this name :)
`JsonValue` makes me expect a Json string. Sure then why is the type not string?

What about `JsonObject` or `JsonModel`.

Just some suggestions and I think as it is also fine.

Reply:
Either of those work for me - I'll update.